### DISCUSSION TITLE
Can't define types inside functions

### DISCUSSION THREAD

**Discussion by Anthony-Gaudino** [vlang/v#11662] [Ideas]
**V version:** 0.1.25 87c0155
**OS:** Ubuntu x86_64

**What did you do?**
Executed this code:

```V
fn main () {
    type Myint int
    a := Myint(100)
}
```

**What did you expect to see?**
Program executed without errors.
 
**What did you see instead?**
```
unexpected token: `}`
    3|     type Myint int
    4|     a := Myint(100)
    5| }
```

--------

Fails with different error:

```V
fn main() {
    type Myint int
    a := Myint(100)
    println(a)
}
```

```
`unresolved` needs to have method `str() string` to be printable
    3|     type Myint int
    4|     a := Myint(100)
    5|     println(a)
                    ^
    6| }
```

--------

This works:

```V
type Myint int                                                                   
                                                                                 
fn main() {                                                                     
    a := Myint(100)                                                              
    println(a)                                                                   
}
```

**Comment by Bowero** [comment]
I'm afraid there could be a lot of problems if you can define types in functions. These could overwrite each other, causing other parts of the program to stop working. I see no added value in supporting this.

**Comment by medvednikov** [comment]
Both C and Go allow this, I've never personally never used this feature.

Any thoughts on the benefits? I guess it allows to define types that are only used in the function, but you can also use unexported module level types.

**Comment by medvednikov** [comment]
One way to handle this is use anonymous functions and structs inside function bodies, and use named types at the module level.

**Comment by Anthony-Gaudino** [comment]
Rust also allows it:

```Rust
fn main() {
    type MyType = u8;
    let a: MyType = 100;

    println!("{:?}", a);
}
```

Here's an example in [Linux](https://github.com/torvalds/linux/blob/b719ae070ee2371c37d846616ef7453ec6811990/arch/s390/include/asm/timex.h#L151)

**Comment by medvednikov** [comment]
It has been decided to support this :)
